File: factors_functions.c
#include <stdio.h>

/**
 * trial_division - Finds the smallest divisor, if any, of a given number `n`
 * @n: The number to factorize
 *
 * This function searches for the smallest divisor of a given number `n`.
 * If a divisor is found, it prints the factorization in the format
 * "n=factor1*factor2". If no divisor is found, it means that `n` is a prime
 * number and it prints "n=n*1".
 *
 * Return: The smallest divisor if found, 0 if `n` is prime.
 */
int trial_division(long int n)
{
long int f;

if (n % 2 == 0)
{
printf("%lu=%lu*%i\n", n, n / 2, 2);
return (0);
}

f = 3;
while (f *f <= n)
{
if (n % f == 0)
{
printf("%lu=%lu*%lu\n", n, n / f, f);
return (0);
}
else
{
f += 2;
}
}
printf("%lu=%lu*%i\n", n, n, 1);
return (0);
}

File: factors_functions.py
#!/usr/bin/python3
import sys
import time
import math


def unix_time(function):
    """
    Return `real`, `sys`, and `user` elapsed time, UNIX 'time' command.
    """
    start_time = time.time()
    function()
    end_time = time.time()

    real_time = end_time - start_time
    user_time = time.process_time()
    sys_time = real_time - user_time

    return f"\nreal: {real_time}\nuser: {user_time}\nsys: {sys_time}"


def trial_division(n: int) -> int:
    """
    Finds the smallest divisor, if any, of a given number `n`
    Returns:
        smallest divisor if found
        0 if n is prime
    """
    if n % 2 == 0:
        print(f"{n}={n // 2}*2")
        return 0

    f = 3
    while f * f <= n:
        if n % f == 0:
            print(f"{n}={n // f}*{f}")
            return 0
        f += 2

    print(f"{n}={n}*1")
    return 0


def print_factors():
    with open(sys.argv[1], 'r') as prime:
        for line in prime:
            n = int(line.strip())
            trial_division(n)


if __name__ == '__main__':
    print(unix_time(print_factors))
